prow/git/git_test.go

```diff
@@ -30,6 +30,8 @@ import (
 	"k8s.io/test-infra/prow/github"
 )
 
+var defaultBranch = localgit.DefaultBranch("")
```

Comment:
Why is `DefaultBranch` always called with the empty string? If using the current directory is always appropriate can we remove the parameter?
Another benefit of this would be that we could determine the default branch once as part of the initialization of the `localgit` package and then expose the value as a global rather than many package level variables. This would be more similar to what Alvaro is suggesting.

Reply:
This is not a global value though. It is something that that can be defined anywhere git can be configured, aka at the system, global, repo and/or submodule levels. 

The change here was not to switch the HEAD branch from master to foo but to make the head branch something configurable, and IMO we should reflect this reality in the code rather than just choose a different global constant.

Reply:
That's also reason I have this as a string: the value of `git config init.defaultBranch` depends on the git repo so allow that to be passed into this exported method (defaulting to the current dir)